packages/experiments/src/components/Button/Button.view.tsx

```diff
@@ -19,40 +19,74 @@ export const ButtonView: IButtonComponent['view'] = props => {
     icon: Icon,
     content: Text,
     menu: ContextualMenu,
-    menuIcon: Icon
+    menuIcon: Icon,
+
+    primaryActionContainer: Stack,
+    secondaryActionContainer: 'span',
+    splitDivider: 'span'
   });
 
   return (
     <Slots.root
       type="button" // stack doesn't take in native button props
       role="button"
+      onClick={split ? undefined : onClick}
       {...buttonProps}
       aria-disabled={disabled}
     >
-      <Slots.stack horizontal as="span" gap={8} verticalAlign="center" horizontalAlign="center" verticalFill>
-        <Slots.icon />
-        <Slots.content />
-        {children}
-        {Menu && (
-          <Stack.Item>
-            <Slots.menuIcon iconName="ChevronDown" />
-          </Stack.Item>
-        )}
-      </Slots.stack>
-      {expanded && Menu && <Slots.menu target={menuTarget} onDismiss={onMenuDismiss} items={[]} />}
+      {split ? renderSplitButton(props, Slots) : renderButton(props, Slots)}
     </Slots.root>
   );
 };
 
-// TODO: test with split button approach.
-//        should split button be another component?
-//        can Button's slots be manipulated to create an HOC split button?
-// {split && (
-// <Slot as='span' userProps={splitContainer}>
-//   <Slot as={Divider} userProps={divider} />
-//   <Slot as={Icon} userProps={menuChevron} />
-// </Slot>
-// )}
+function renderButton(props: IPropsWithChildren<IButtonViewProps>, Slots: ISlots<Required<IButtonSlots>>): JSX.Element {
```

Comment:
Hmm, I think there's an opportunity to create a render helper type in Foundation, like maybe:
```tsx
function renderButton: ISlotRenderHelper<IButtonViewProps, IButtonSlots> = (props, Slots) => {
  ...
}
```